### ISSUE TITLE
App Crashes on DB Unlock on Mac Catalina (10.15.7)

### ISSUE THREAD

**Issue by medixcare** [keepassxreboot/keepassxc#7760]
## Overview
Starting today the app crashes once I unlock it. Using a wrong password triggers the normal error messages and the app does not crash. Initially the crash occurred when trying to unlock via fingerprint so I restarted the machine. Sadly nothing changed.

## Steps to Reproduce
[NOTE]: # ( Provide a simple set of steps to reproduce this bug. )
1.  KeePassXC 2.7.0 starts on system start
2. Enter password for database
3. click "Unlock"
4. App crashed

## Expected Behavior
App does not crash and allows access to passwords

## Actual Behavior
App crashes. Passwords inaccessible

## Context
The db makes use of the KeyShare feature and has multiple shared dbs.
I have to expect a specific password to be the cause as this setup was in use for some days and without a change on my part the app started crashing on unlock.
DB files are stored in a directory that is getting synced via "SynologyDrive" to our NAS.
Some other Issues I read that don't really match this problem:

- #3942 
no random crashes. only at unlock
- #3857
could be related but I'm not able to open the database. Saving general settings seems to work
- #4398
no crashes on app start. just at unlock

[NOTE]: # ( Paste debug info from Help → About here )
KeePassXC - Version 2.7.0
Revision: d7a9ef4

Qt 5.15.2
Diagnostic mode is deactivated.

Operating System: macOS 10.15
CPU-Architecture: x86_64
Kernel: darwin 19.6.0

Active Extensions:
- Auto-Type
- Browser-Integration
- SSH-Agent
- KeeShare
- YubiKey
- Fast Unlock (deactivating this did not help!)

Cryptographic Library:
- Botan 2.19.1

[NOTE]: # ( Pick choices based on your environment )
Operating System: macOS
Desktop Env: System default (I'm actually not sure what macOs uses)

**Comment by medixcare** [comment]
Tried opening the same file at a colleagues computer running ubuntu 20.04 and keepassxc 2.4.3 and the db can be opened without problems.

**Comment by droidmonkey** [comment]
Can you provide a stack trace / crash report please? You can also try a snapshot at https://snapshot.keepassxc.org. 

Finally you can downgrade to 2.6.6 to restore your access. Go to our releases page to download the old version.

**Comment by medixcare** [comment]
I downgraded to 2.6.6 and first I did not accept the signatures for the shares. With this I was able to open the file again.
Then step-by-step I accepted all the signatures and found it working. But that was followed be keepass wanting to combine databases as me not accepting the signatures must have caused a diversion in the database. (This is just me trying to make sense of what happend)
The result was an unaccessible database (again).

Today I upgraded to 2.7.1 and was able to open the file again without any problems!

So I'm happy to be able to work with the keepass db again. Sadly I don't have additional info to pinpoint the initial cause. Hopefully the actual cause was fixed and not just some side effect.

thank you for your great work!
